backend/shared/src/create-notification.ts

```diff
@@ -1562,3 +1562,27 @@ export const createBountyAddedNotification = async (
   const pg = createSupabaseDirectClient()
   await insertNotificationToSupabase(notification, pg)
 }
+
+export const createCustomNotification = async (
+  recipientId: string,
+  title: string,
+  sender: User,
+  url?: string
+) => {
+  const notification: Notification = {
+    id: crypto.randomUUID(),
+    userId: recipientId,
+    reason: 'announcement_created',
+    createdTime: Date.now(),
+    isSeen: false,
+    sourceId: '',
+    sourceType: 'announcement',
+    sourceUserName: sender.name,
+    sourceUserUsername: sender.username,
+    sourceUserAvatarUrl: sender.avatarUrl,
+    sourceText: title,
+    data: { url },
+  }
+  const pg = createSupabaseDirectClient()
```

Comment:
Don't you have to check somewhere if they've opted out of browser notifications? See the line `if (userOptedOutOfBrowserNotifications(privateUser)) return` from just above your function.